ENGDOCS-2305 (#21397)
docker/docker.github.io: content/manuals/security/for-admins/hardened-desktop/settings-management/_index.md

weight: 10
---

> [!NOTE]
>
> Settings Management is available to Docker Business customers only.

> Settings Management helps you control key Docker Desktop settings, like proxies and network configurations, on your developer's machines within your organization.

Reviewer: ```suggestion
Settings Management helps you control key Docker Desktop settings, like proxies and network configurations, on your developers' machines within your organization.
```

> For an extra layer of security, you can also use Settings Management to enable and lock in [Enhanced Container Isolation](../enhanced-container-isolation/_index.md) which ensures that any configurations set with Settings Management cannot be modified by containers.

Reviewer: ```suggestion
For an extra layer of security, you can also use Settings Management to enable and lock in [Enhanced Container Isolation](../enhanced-container-isolation/_index.md), which prevents containers from modifying any Settings Management configurations.
```

Reviewer: small grammar/readability nit and a reword for active voice

## How do I set up and enforce Settings Management?

You first need to [enforce sign-in](/manuals/security/for-admins/enforce-sign-in/_index.md) to ensure that all Docker Desktop developers authenticate with your organization. Since the Settings Management feature requires a Docker Business subscription, enforced sign-in guarantees that only authenticated users have access and that the feature consistently takes effect across all users, even though it may still work without enforced sign-in.

Next, you must either manually [create and configure the admin-settings.json file](configure.md), or use the `--admin-settings` installer flag on [macOS](/manuals/desktop/setup/install/mac-install.md#install-from-the-command-line) or [Windows](/manuals/desktop/setup/install/windows-install.md#install-from-the-command-line) to automatically create the `admin-settings.json` and save it in the correct location.

Once this is done, Docker Desktop developers receive the changed settings when they either:
- Quit, re-launch, and sign in to Docker Desktop
- Launch and sign in to Docker Desktop for the first time

> So as not to disrupt your developers' workflow, Docker doesn't automatically mandate that developers re-launch and re-authenticate once a change has been made.

Reviewer: ```suggestion
To avoid disrupting your developers' workflow,, Docker doesn't automatically require that developers re-launch and re-authenticate once a change has been made.
```

## What do developers see when the settings are enforced?

> Enforced settings in Docker Desktop appear grayed out, making them uneditable through the Docker Desktop Dashboard, CLI, or the `settings-store.json` file (or `settings.json` for Docker Desktop versions 4.34 and earlier).

Reviewer: ```suggestion
Enforced settings appear grayed out in Docker Desktop. They can't be edited via the Docker Desktop Dashboard, CLI, or `settings-store.json` (or `settings.json` for Docker Desktop 4.34 and earlier).
```

Reviewer: reword to fix vale error about `uneditable`

> In addition, if Enhanced Container Isolation is enforced, developers can't use privileged containers or similar techniques to modify enforced settings within the Docker Desktop Linux VM, for example, reconfigure proxy and networking of reconfigure Docker Engine.

Reviewer: ```suggestion
In addition, if Enhanced Container Isolation is enforced, developers can't use privileged containers or similar techniques to modify enforced settings within the Docker Desktop Linux VM. For example, they can't reconfigure proxy and networking, or Docker Engine.
```

Reviewer: not sure if the `of` in the last part of this section was supposed to be `or`, so update as needed!
